src/pkg/packager/creator/differential.go

```diff
@@ -0,0 +1,118 @@
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-FileCopyrightText: 2021-Present The Zarf Authors
+
+// Package creator contains functions for creating Zarf packages.
+package creator
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/defenseunicorns/zarf/src/config"
+	"github.com/defenseunicorns/zarf/src/internal/packager/git"
+	"github.com/defenseunicorns/zarf/src/pkg/layout"
+	"github.com/defenseunicorns/zarf/src/pkg/message"
+	"github.com/defenseunicorns/zarf/src/pkg/packager/sources"
+	"github.com/defenseunicorns/zarf/src/pkg/transform"
+	"github.com/defenseunicorns/zarf/src/pkg/utils"
+	"github.com/defenseunicorns/zarf/src/types"
+	"github.com/go-git/go-git/v5/plumbing"
+)
+
+// loadDifferentialData sets any images and repos from the existing reference package in the DifferentialData and returns it.
+func (pc *PackageCreator) loadDifferentialData() (diffData *types.DifferentialData, err error) {
+	diffPkgPath := pc.createOpts.DifferentialData.DifferentialPackagePath
+	if diffPkgPath != "" {
+		pc.pkgOpts.PackageSource = diffPkgPath
+	}
+
+	tmpdir, err := utils.MakeTempDir(config.CommonOptions.TempDirectory)
+	if err != nil {
+		return nil, err
+	}
+
+	diffLayout := layout.New(tmpdir)
+	defer os.RemoveAll(diffLayout.Base)
+
+	src, err := sources.New(pc.pkgOpts)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
simply make a fresh `types.PkgOpts` struct here, dont pass down and set onto `PackageCreator`